lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -519,6 +566,69 @@ pcmk_designated_controller(xmlNodePtr *xml, unsigned int message_timeout_ms)
     return rc;
 }
 
+/*!
+ * \internal
+ * \brief Get the node name corresponding to the specified node ID
+ *
+ * \param[in,out] out                 Output object
+ * \param[in]     node_id             ID of node whose name to get (or 0 for
+ *                                    local node)
+ * \param[out]    node_name           If not \p NULL, where to store the node
+ *                                    name
+ * \param[in]     show_output         Whether to show the node name
+ * \param[in]     message_timeout_ms  How long to wait for a reply from the
+ *                                    \p pacemaker-controld API. If 0,
+ *                                    \p pcmk_ipc_dispatch_sync will be used.
+ *                                    Otherwise, \p pcmk_ipc_dispatch_poll will
+ *                                    be used.
+ *
+ * \return Standard Pacemaker return code
+ *
+ * \note The caller is responsible for freeing \p node_name using \p free().
+ */
+int
+pcmk__node_name(pcmk__output_t *out, uint32_t node_id, char **node_name,
+                bool show_output, unsigned int message_timeout_ms)
```

Comment:
Is this too likely to get confused with `pe__node_name()`? I prefer the compact name, so hopefully the namespace difference is enough.

Reply:
Oh that's a good idea.

Reply:
Or we could go a step further and make this a `pcmk__query_node_info()` function that takes a four more [out] args arguments and returns everything from `pcmk_controld_api_node_info()` (see below).

```
static void
set_node_info_data(pcmk_controld_api_reply_t *data, xmlNode *msg_data)
{       
    data->reply_type = pcmk_controld_reply_info;
    if (msg_data == NULL) {
        return;
    }   
    data->data.node_info.have_quorum = pcmk__xe_attr_is_true(msg_data, XML_ATTR_HAVE_QUORUM);
    data->data.node_info.is_remote = pcmk__xe_attr_is_true(msg_data, XML_NODE_IS_REMOTE);
    crm_element_value_int(msg_data, XML_ATTR_ID, &(data->data.node_info.id));
    data->data.node_info.uuid = crm_element_value(msg_data, XML_ATTR_UUID);
    data->data.node_info.uname = crm_element_value(msg_data, XML_ATTR_UNAME);
    data->data.node_info.state = crm_element_value(msg_data, XML_NODE_IS_PEER);
}   
```

Reply:
It's internal, so no big deal either way. The public API call will return the info as XML so no real effect on that.

Reply:
Planning to do a `pcmk__query_node_info()` and keep messages for default and XML that show all the info items. If we want we can eventually add a `pcmk__query_node_name()` that acts as a wrapper.